### PR TITLE
365 image uploads

### LINKED ISSUE #365: Profile image uploading
Now that we have S3 ability on alpha, we can properly test the profile upload flow.

It appears to work, but it takes some time for the image to upload and become processed to the extent it shows up in-app. We don't show any loading indication or anything like that, and it feels broken because of that.

How can we fix?

### PR DESCRIPTION
**Pull request by Lepozepo**
Fixes #365 

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer

## Reviewer

- [x] Run `test` and make sure all tests pass
- [x] Review function and form on iOS
- [x] Review function and form on Android
- [x] Review function and form on Web
- [x] Review new test logic

### CODE REVIEW

**Comment by Lepozepo** [comment]
@burkeshartsis I'm not sure if this will fix it (something changed in rock that I'm not able to tap into it anymore from my local Heighliner) but it makes sense that it would.

**Comment by IsaacHardy** [comment]
@Lepozepo We made some changes to the env structure so we create new PR to the alpha branch now.

**Comment by burkeshartsis** [comment]
@IsaacHardy I'll take care of moving the PR.

**Comment by burkeshartsis** [comment]
Alright I think this is pointed in the right direction now. I'll work on getting the rest of the UI for this later  today/tomorrow morning.

**Comment by IsaacHardy** [comment]
@burkeshartsis Looks like I'm running into an issue while testing.

It seems I receive a 401 unauthorized when attempting to upload a file. I tested on beta and alpha but received this error on both environments.

**Comment by burkeshartsis** [comment]
@IsaacHardy so I was getting intermittent issues too and when I switched branches it didn't seem to change so I figured the problem was environment based. I'll look into it.

File: src/@utils/makeCancelable.js [hunk 1]
```diff
@@ -3,8 +3,8 @@ const makeCancelable = (promise) => {
 
   const wrappedPromise = new Promise((resolve, reject) => {
     promise.then(
-      val => hasCanceled ? reject({isCanceled: true}) : resolve(val), // eslint-disable-line
-      error => hasCanceled ? reject({isCanceled: true}) : reject(error), // eslint-disable-line
+      val => hasCanceled ? reject({isCancelled: true}) : resolve(val), // eslint-disable-line
```
Comment (burkeshartsis) [inline src/@utils/makeCancelable.js]: How very British of you @Lepozepo

File: src/@ui/forms/UploadProfileImageForm/index.js [hunk 2]
```diff
@@ -7,29 +8,51 @@ import fetch from '@utils/fetch';
...
         });
 
+        if (props.onUploadEnded) {
+          props.onUploadEnded();
+        }
+
+        // NOTE: Refetch user to update current user state
+        // and trigger any necessary rerenders
+        // (not tested yet @burke)
+        await props.refetch();
```
Comment (Lepozepo) [inline src/@ui/forms/UploadProfileImageForm/index.js]: @burkeshartsis I think you can remove `await`

Comment (burkeshartsis) [reply src/@ui/forms/UploadProfileImageForm/index.js]: yeah that didn't fix the upload problem. It keeps throwing the `catch` statement.

**Comment by burkeshartsis** [comment]
@IsaacHardy so I finally got all the enviroment issues aligned to the point where this PR can be tested and merged. If everything is pointed to `beta-rock` and `beta-api` image uploading works.

**Comment by IsaacHardy** [comment]
@burkeshartsis I am using all beta environments to test and still cannot seem to upload an image on the profile. I'm gonna check to make sure beta image uploading works.

**Comment by IsaacHardy** [comment]
@burkeshartsis Beta image uploading api is working. Could you walk me through the flow for testing this PR?

**Pull request merged**